mlflow/store/tracking/file_store.py

```diff
@@ -1059,6 +1070,118 @@ def record_logged_model(self, run_id, mlflow_model):
         except Exception as e:
             raise MlflowException(e, INTERNAL_ERROR)
 
+    def log_inputs(self, run_id: str, datasets: List[DatasetInput]):
+        """
+        Log inputs, such as datasets, to the specified run.
+
+        :param run_id: String id for the run
+        :param datasets: List of :py:class:`mlflow.entities.DatasetInput` instances to log
+                         as inputs to the run.
+
+        :return: None.
+        """
+        _validate_run_id(run_id)
+        run_info = self._get_run_info(run_id)
+        check_run_is_active(run_info)
+        experiment_dir = self._get_experiment_path(run_info.experiment_id, assert_exists=True)
+        run_dir = self._get_run_dir(run_info.experiment_id, run_id)
+
+        for dataset_input in datasets:
+            dataset = dataset_input.dataset
+            dataset_id = FileStore._get_dataset_id(
+                dataset_name=dataset.name, dataset_digest=dataset.digest
+            )
+            dataset_dir = os.path.join(experiment_dir, FileStore.DATASETS_FOLDER_NAME, dataset_id)
+            if not os.path.exists(dataset_dir):
+                os.makedirs(dataset_dir, exist_ok=True)
+                write_yaml(dataset_dir, FileStore.META_DATA_FILE_NAME, dict(dataset))
+
+            input_id = FileStore._get_input_id(dataset_id=dataset_id, run_id=run_id)
+            input_dir = os.path.join(run_dir, FileStore.INPUTS_FOLDER_NAME, input_id)
+            if not os.path.exists(input_dir):
+                os.makedirs(input_dir, exist_ok=True)
+                fs_input = FileStore._FileStoreInput(
+                    source_type=FileStore.INPUT_VERTEX_TYPE_DATASET,
+                    source_id=dataset_id,
+                    destination_type=FileStore.INPUT_VERTEX_TYPE_RUN,
+                    destination_id=run_id,
+                    tags=dataset_input.tags,
+                )
+                write_yaml(input_dir, FileStore.META_DATA_FILE_NAME, fs_input._asdict())
```

Comment:
From https://stackoverflow.com/a/26180604/11952869 , this method is stable and safe to call. It's just prefixed with `_` so that users could still put a field called `asdict` in their namedtuple (avoids conflict)